mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala

```diff
@@ -153,19 +153,27 @@ object AlignmentRecordMaterialization extends Logging {
    * @return RDD of data from the file over specified ReferenceRegion
    */
   def load(sc: SparkContext, fp: String, region: Option[ReferenceRegion]): AlignmentRecordRDD = {
-    if (fp.endsWith(".adam")) loadAdam(sc, fp, region)
-    else {
-      try {
-        AlignmentRecordMaterialization.loadFromBam(sc, fp, region)
-          .transform(rdd => rdd.filter(_.getReadMapped))
-      } catch {
-        case e: Exception => {
-          val sw = new StringWriter
-          e.printStackTrace(new PrintWriter(sw))
-          throw UnsupportedFileException("bam index not provided. Stack trace: " + sw.toString)
+    val alignments =
+      if (fp.endsWith(".adam")) loadAdam(sc, fp, region)
+      else {
+        try {
+          AlignmentRecordMaterialization.loadFromBam(sc, fp, region)
+            .transform(rdd => rdd.filter(_.getReadMapped))
+        } catch {
+          case e: Exception => {
+            val sw = new StringWriter
+            e.printStackTrace(new PrintWriter(sw))
+            throw UnsupportedFileException("bam index not provided. Stack trace: " + sw.toString)
+          }
         }
       }
-    }
+    alignments.transform(rdd => rdd.map(r => {
+      // if read group is empty, fill it
+      // see requirements in GA4GHConverter.scala
+      if (r.getRecordGroupName() == null)
+        r.setRecordGroupName("N/A")
```

Comment:
Perhaps the default-if-missing should go in the GA4GH converter then, say an Option or else here instead of a require

https://github.com/bigdatagenomics/mango/blob/master/mango-core/src/main/scala/org/bdgenomics/mango/converters/GA4GHConverter.scala#L72